src/Events/Transaction.php

```diff
@@ -45,26 +52,38 @@ class Transaction extends TraceableEvent implements \JsonSerializable
     private $backtraceLimit = 0;
 
     /**
-    * Create the Transaction
-    *
-    * @param string $name
-    * @param array $contexts
-    */
-    public function __construct(string $name, array $contexts, $start = null)
+     * Create the Transaction
+     *
+     * @param string $name
+     * @param array $contexts
+     * @param TimerFactory|null $timerFactory
+     */
+    public function __construct(string $name, array $contexts, TimerFactory $timerFactory = null)
     {
         parent::__construct($contexts);
         $this->setTransactionName($name);
-        $this->timer = new Timer($start);
+        $this->timerFactory = $timerFactory ?? new TimerFactory();
     }
 
     /**
-    * Start the Transaction
-    *
-    * @return void
-    */
-    public function start()
+     * Start the Transaction
+     *
+     * @param float|null $startTime
+     * @return void
+     * @throws AlreadyStartedException
+     */
+    public function start(float $startTime = null)
     {
-        $this->timer->start();
+        if (null !== $this->timer) {
+            throw new AlreadyStartedException();
+        }
+
+        $this->timer = $this->timerFactory->newTimer($startTime);
+        // Start the timer only if there was not a start time
+        if (null === $startTime) {
```

Comment:
If we add a `isStarted` method to `Timer`, we could replace the `if` statement by that method. It's more natural IMO.

Reply:
That would work, but I intend to redo the enter timing implementation in the next release. I've grown to dislike having new `Timer` objects created for each event and does not feel right to create a `Timer` with a specific start time. I'm not clear what condition I'm trying to express by doing so. I understand why it's necessary right now. It's the only way to take an event for which you know the start time and duration, but which executes outside of your control, and add it to the `Transaction`. 

I believe that's a misuse of a `Timer` and why I want to separate the timing of an event from the creation of an event. A `TimedEvent` (or something) should be created with a timestamp and duration already set. Something else does the timing and makes that data available. That decouples how the timing is done from the representation of the event.

So in this particular case, I'm still making the minimum change I can to the fewest classes to fix the immediate problem. An `isStarted` method on the `Timer` would make this clearer, though, so I'll make that change.

Reply:
I added both `Timer::isStarted()` and `Timer::isNotStarted` since the latter made more semantic sense in our use case.